src/app/users/user-edit/user-edit.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DataService } from 'src/app/shared/data-service';
+import { IUser } from '../user.interface';
+
+@Component({
+  selector: 'app-user-edit',
+  templateUrl: './user-edit.component.html',
+  styleUrls: ['./user-edit.component.scss'],
+})
+export class UserEditComponent implements OnInit {
+  id!: number;
```

Comment:
the way you implemeted this form, don't looks cool.  you do a mix with ngmode and a form.

Change all this.  Implement Reactive Form.
Take advantage of formBuilder.
 study read all this, especially the amazing angular formBuilder.
https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls

Reply:
Please,  do this Point.... the code looks like you did not see this comment.